etl_manager/etl.py

```diff
@@ -385,12 +404,54 @@ def job_run_state(self) :
 
         return status['JobRun']['JobRunState'].lower()
 
+    def wait_completion(self):
+        """
+        Wait for the job to complete.
+
+        This means it either succeeded or it was manually stopped.
+
+        Raises:
+            JobFailed: When the job failed
+            JobTimedOut: When the job timed out
+        """
+
+        while True:
```

Comment:
boto already has a mechanism for polling until something is done:

you could use a boto waiter instead of a loop/poll (it does that under the hood anyway)
https://boto3.readthedocs.io/en/latest/reference/services/glue.html#Glue.Client.get_waiter

but don't spend too long on it if you can't get it working straight away

Reply:
some examples: https://stackoverflow.com/questions/40969984/how-to-use-boto3-waiters-to-take-snapshot-from-big-rds-instances

Reply:
Ignore above comment! 
I just checked the supported waiters:

    >>> 
    >>> _glue_client = boto3.client('glue', 'eu-west-1')
    >>> _glue_client.waiter_names
    []
    >>> 
and it looks like glue isn't supported?

Reply:
🍝 I'll have a look but I already don't understand the docs 🤣

Reply:
yeah the docs are terrible but if there are no `.waiter_names` in your client then it won't work so we can't use it here anyway!

Reply:
👍